swig/cpp/tests/changes.cpp

```diff
@@ -0,0 +1,219 @@
+                auto it = sess->get_changes_iter(&change_path[0]);
+
+                while (auto change = sess->get_change_next(it)) {
+                    assert(SR_OP_DELETED == change->oper());
+                }
+            } catch (const std::exception& e) {
+                cout << e.what() << endl;
+            }
+            return SR_ERR_OK;
+        }
+};
+
+void
+test_module_change_delete(S_Session sess)
+{
+    S_Subscribe subs(new Subscribe(sess));
+    S_Callback cb(new DeleteCb());
+
+    init_test(sess);
+    usleep(100);
+
+    subs->module_change_subscribe(module_name, cb, NULL, 0, SR_SUBSCR_DEFAULT | SR_SUBSCR_APPLY_ONLY);
+
+    const char *xpath = (char*)get_xpath(get_test_name(LOW_BOUND), (char *) "number");
+    S_Val v = sess->get_item(xpath);
+    sess->delete_item(xpath);
+    sess->commit();
+
+    usleep(100);
+    subs->unsubscribe();
+    usleep(100);
+}
+
+class ModifyCb: public Callback {
+public:
+    int module_change(S_Session sess, const char *module_name, sr_notif_event_t event, void *private_ctx)
+        {
+            char change_path[MAX_LEN];
+
+            try {
+                snprintf(change_path, MAX_LEN, "/%s:*", module_name);
+                auto it = sess->get_changes_iter(&change_path[0]);
+
+                while (auto change = sess->get_change_next(it)) {
+                    assert(SR_OP_MODIFIED == change->oper());
+                }
+            } catch (const std::exception& e) {
+                cout << e.what() << endl;
+            }
+            return SR_ERR_OK;
+        }
+};
+
+void
+test_module_change_modify(S_Session sess)
+{
+    S_Subscribe subs(new Subscribe(sess));
+    S_Callback cb(new ModifyCb());
+
+    init_test(sess);
+    usleep(100);
+
+    subs->module_change_subscribe(module_name, cb, NULL, 0, SR_SUBSCR_DEFAULT | SR_SUBSCR_APPLY_ONLY);
+
+    const char *xpath = (char*)get_xpath(get_test_name(LOW_BOUND), (char *) "number");
+    S_Val vset(new Val((int32_t)42, SR_INT32_T));
+    sess->set_item(xpath, vset);
+    sess->commit();
+
+    usleep(100);
+    subs->unsubscribe();
+    usleep(100);
+}
+
+class CreateCb: public Callback {
+public:
+    int module_change(S_Session sess, const char *module_name, sr_notif_event_t event, void *private_ctx)
+        {
+            char change_path[MAX_LEN];
+
+            try {
+                snprintf(change_path, MAX_LEN, "/%s:*", module_name);
+                auto it = sess->get_changes_iter(&change_path[0]);
+
+                while (auto change = sess->get_change_next(it)) {
+                    assert(SR_OP_CREATED == change->oper());
+                }
+            } catch (const std::exception& e) {
+                cout << e.what() << endl;
+            }
+            return SR_ERR_OK;
+        }
+};
+
+void
+test_module_change_create(S_Session sess)
+{
+    S_Subscribe subs(new Subscribe(sess));
+    S_Callback cb(new CreateCb());
+
+    init_test(sess);
+    usleep(100);
```

Comment:
What's the purpose of these usleeps (also two more below)? How was the value determined? I'm asking because I'm getting several failures from the test suite in our internal code which uses sysrepo and multicore CI VMs. If there's any threading involved here and the test cannot be redesigned, then the timeouts must really be longer. If there's no threading, why bother sleeping?